### PR TITLE
issue #22 - support dynamic configs

### PR DESCRIPTION
**Pull request by tjstansell**
This adds support for issue #22 config plugins that can dynamically adjust the running config.  While this technically doesn't rely on any other PRs, it is most useful after PR #21 and the included mysql-config config plugin doesn't make sense without PR #23.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -112,6 +112,9 @@ ehthumbs_vista.db
 # Folder config file
 Desktop.ini
 
+# VIM tempfiles
+*.swp
```
Comment (wolph) [inline .gitignore]: In that case we probably need `*.swo` as well :)

**Comment by wolph** [comment]
This pull request looks great, thanks for all of the help! I don't see any issues with it but I should note that I've only reviewed the code, I haven't tested it

**Comment by Rockstar04** [comment]
@tjstansell would you be able to resolve the conflicts with master now that #21 and #23 have been merged?

**Comment by tjstansell** [comment]
Yeah.  I'll clean up the merge issues and update the PR.

File: .gitignore (hunk 1 above)
Comment (tjstansell) [reply .gitignore]: I'll update this PR with *.swo and *~ as well ... i never have more than one swap file, so I wasn't aware of .swo :)

File: newrelic_python_agent/agent.py [hunk 2]
```diff
@@ -104,49 +108,113 @@ def license_key(self):
...
+            else:
+                thread = threading.Thread(target=self.thread_metric_process,
+                                          kwargs={'config': instance,
+                                                  'name': instance_name,
+                                                  'plugin': plugin,
+                                                  'poll_interval':
+                                                      int(self._wake_interval)})
+            LOGGER.info("Starting plugin instance %s as thread %s", instance_name, thread.getName())
+            self.thread_names[instance_name] = thread.getName()
+            thread.run()
```
Comment (tjstansell) [inline newrelic_python_agent/agent.py]: argh.  looks like i messed up the rebase of this PR and accidentally reverted the use_threading PR.  I'll fix this.

**Comment by Rockstar04** [comment]
I'm inline with @WoLpH. The PR looks great, but I haven't personally tested it.

**Pull request merged**